crates/client/rpc/src/lib.rs

```diff
@@ -1162,12 +1146,15 @@ where
             FieldElement::default()
         };
 
-        let starknet_block_hash = BlockHash(starknet_block.header().hash().into());
+        let block_transactions = starknet_block.transactions();
 
-        let state_diff = self.get_state_diff(&starknet_block_hash).map_err(|e| {
-            error!("Failed to get state diff. Starknet block hash: {starknet_block_hash}, error: {e}");
-            StarknetRpcApiError::InternalServerError
-        })?;
+        let previous_block_substrate_hash = get_previous_block_substrate_hash(self, substrate_block_hash)?;
+
+        let state_diff = self.get_transaction_re_execution_state_diff(
+            previous_block_substrate_hash,
+            vec![],
+            block_transactions.clone(),
```

Comment:
Unnecessary clone, no?
Try running clippy

Reply:
it returns reference, so as i guess it can't be moved